jupyterlab_server/handlers.py

```diff
@@ -24,6 +24,8 @@
 # -----------------------------------------------------------------------------
 
 
+MASTER_URL_PATTERN = '/(?P<mode>lab|doc)(?P<workspace>/workspaces/[a-zA-Z0-9\-\_]+)?(?P<tree>/tree/.*)?'
```

Comment:
We had intentionally made `lab` in the url configurable in case folks have a different name for their application.

Reply:
I think we should be okay with deployers/admins owning the server, port, and `base_url`, and deciding that all the paths that get routed by the application server after `base_url` belong to us, but I did wonder if we should make this be a major version bump because even if the programmatic API is backward-compatible, URL decisions are also API from my point of view.

Reply:
With this change the `/lab` portion formally becomes part of our API, and this is backwards incompatible. We definitely need a major release for this. Agree with what @afshin is saying here that the server, port and base_url are the configurable portion.